client/keyspace_client.go

```diff
@@ -111,3 +113,44 @@ func (c *client) WatchKeyspaces(ctx context.Context) (chan []*keyspacepb.Keyspac
 	}()
 	return keyspaceWatcherChan, err
 }
+
+// UpdateKeyspaceState attempts to update the keyspace specified by ID to the target state,
+// it will also record StateChangedAt for the given keyspace if a state change took place.
+// Currently, legal operations includes:
+//
+//	ENABLED -> {ENABLED, DISABLED}
+//	DISABLED -> {ENABLED, DISABLED, ARCHIVED}
+//	ARCHIVED -> {ARCHIVED, TOMBSTONE}
```

Comment:
What the different between the `{ARCHIVED, TOMBSTONE}`?

Reply:
Archive is used to denote that keyspace's corresponding data in store nodes are safe to remove/ currently being removed (or being archived into cold storage somewhere else), a point of no return for the keyspace (as oppose to disabled which means data are still intact). This tells gc worker to keep cleaning up data belong to this keyspace.

Tombstone is used to denote that keyspace's data in storage has been completely removed, gc worker can safely skip gcing this keyspace